Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -6,18 +6,63 @@
  *
  */
 
+#include <QMessageBox>
 #include <QPushButton>
 #include <QVBoxLayout>
 
 #include "RobotImporter/RobotImporterWidget.h"
-#include "RobotImporter/URDF/URDFPrefabMaker.h"
+
+#include <AzCore/Utils/Utils.h>
 
 namespace ROS2
 {
+    namespace Internal
+    {
+        AZStd::optional<QString> GetPathWithExtension(
+            const AZStd::string& extensionDescription, QFileDialog::FileMode mode, QWidget* parent = nullptr)
+        {
+            QFileDialog importFileDialog(parent);
+            importFileDialog.setDirectory(AZ::Utils::GetProjectPath().c_str());
+            importFileDialog.setFileMode(mode);
+            importFileDialog.setNameFilter(QObject::tr(extensionDescription.c_str()));
+            importFileDialog.setViewMode(QFileDialog::Detail);
+
+            int result = importFileDialog.exec();
+            if (result != QDialog::DialogCode::Accepted)
+            {
+                return AZStd::nullopt;
+            }
+
+            return importFileDialog.selectedFiles().first();
+        }
+
+        ExistingPrefabAction GetExistingPrefabAction(QWidget* parent = nullptr)
+        {
+            QMessageBox msgBox(parent);
+            msgBox.setWindowTitle(QObject::tr("Prefab file exists"));
+            msgBox.setText(QObject::tr("The prefab file already exists."));
```

Comment:
add <name> to debug - so the user knows which prefab file, unless it is obvious in the context